### DISCUSSION TITLE
Purge doesn't keep tailwind classes used in HTML

### DISCUSSION THREAD

**Discussion by noook** [tailwindlabs/tailwindcss#2703] [Help] [answered]
### Describe the problem:

**EDIT**: The output is a library, not an app (entrypoint is not the same)

The classes I added in the HTML are not conserved when building and purging style.

There is no CSS output
```vue
<template>
  <div class="text-3xl text-red-100">
    <h1>Hello</h1>
  </div>
</template>
```
```js
module.exports = {
  purge: {
    content: [
      './src/**/*.vue',
      './src/**/*.html',
      './src/**/*.js',
    ],
  },
};
```

### Link to a minimal reproduction:

[Github repository](https://github.com/noook/vue-ui-lib)

`yarn build` to see that there is no CSS included in the build

**Comment by sa3dany** [comment]
Are you sure postcss runs? I've checked the build script in package.json and the only thing it does is execute `vue-cli-service build`.

**Comment by noook** [comment]
Isn't PostCSS part of the build process ? If I change the purge config or put wrong configuration I have warnings during the biuld process so I guess the purge step is executed.

**Comment by sa3dany** [comment]
I am not really familiar with `vue-cli` so that's why I asked. Does running postcss directly also not produce an output?
`postcss src/assets/main.scss --output dist/styles.css`

**Comment by noook** [comment]
It runs correctly, it outputs a file of 106 347 lines, I guess nothing's purged in it.

**Comment by sa3dany** [comment]
You need to set the NODE_ENV to production for the purging to happen `NODE_ENV=production postcss src/assets/main.scss --output dist/styles.css`.

**Comment by noook** [comment]
Oh right, indeed it works way better. And my classes are now appended to the output, which is nice ! What's the trick then ? 

I checked `vue-cli` documentation under the "Working with postcss" section, so my config file should be read when building.

Any idea what's happening ? That's the first time I work with a post-processor I don't think I have enough knowledge about those things yet

**Comment by sa3dany** [comment]
Adding `import './assets/main.scss';` in `src/App.vue` within the <script> tags seems to produce the correct output.
I suggest closing this issue though since we have established that this is not a tailwind issue.

**Comment by Destreyf** [comment]
@noook @sa3dany 

I am using tailwind with vue in one of my projects, I imported the base style into my main.ts file and it worked as well.

**Reply by noook** [reply]
yeah it transforms every `@apply` rule into compiled CSS, but it doesn't keep the classes I'm using on my markup.

**Comment by simonswiss** [comment]
The default strategy to decide wether or not to run `purge` in Tailwind,  like @sa3dany [mentioned above](https://github.com/tailwindlabs/tailwindcss/discussions/2703#discussioncomment-116922), is based on the `NODE_ENV === "production"` check.

I see you have now [added this in your package.json script](https://github.com/noook/vue-ui-lib/blob/master/package.json#L7), which should make it work properly.

If you want to handle that differently, you can [take control of when to purge](https://tailwindcss.com/docs/controlling-file-size#enabling-manually) by adding an `enabled` property to your `purge` object in the Tailwind config file, where you can decide of the logic:

```diff
module.exports = {
  purge: {
    content: [
      './src/**/*.vue',
      './src/**/*.html',
      './src/**/*.js',
    ],
+.  enabled: true // this would *always* run purge
  },
};
```

Hope this helps! 👍

**Reply by noook** [reply]
That's a good point, I'll rather do this indeed.

**Comment by noook** [comment, marked as answer]
Alright alright, I figured it out. The build command doesn't seem to take `main.ts` as the entry point when **outputting as a library**, `App.vue` is. I moved the import instruction into the directory I'm building the library from and it works like a charm. Fixed in [this commit](https://github.com/noook/vue-ui-lib/commit/c2ae9eeff86a35c22f38f7835dc14dfa59280d06)

Special thanks to @sa3dany for [leading me on the good path](https://github.com/tailwindlabs/tailwindcss/discussions/2703#discussioncomment-116924)

I'll update my original post to mention that the output is a library and not an app, and that the entry point is not the one we all think about.
